src/liballoc/str.rs

```diff
@@ -377,12 +377,15 @@ impl str {
 
     /// Escapes each char in `s` with [`char::escape_debug`].
     ///
+    /// Note: only extended grapheme codepoints that begin the string will be
+    /// escaped.
+    ///
     /// [`char::escape_debug`]: primitive.char.html#method.escape_debug
     #[unstable(feature = "str_escape",
                reason = "return type may change to be an iterator",
                issue = "27791")]
     pub fn escape_debug(&self) -> String {
-        self.chars().flat_map(|c| c.escape_debug()).collect()
+        self.chars().enumerate().flat_map(|(i, c)| c.escape_debug_ext(i == 0)).collect()
```

Comment:
Please write this as follow to avoid counting code points:

```rust
    let mut string = String::with_capacity(self.len());
    let mut chars = self.chars();
    if let Some(first) = chars.next() {
        string.extend(first.escape_debug_ext(true))
    }
    string.extend(chars.flat_map(|c| c.escape_debug_ext(false)));
    string
```